dipy/align/imaffine.py

```diff
@@ -222,8 +225,9 @@ def _apply_transform(self, image, interp='linear', image_grid2world=None,
         shape = np.array(sampling_grid_shape, dtype=np.int32)
 
         # Verify valid image dimension
-        if dim < 2 or dim > 3:
-            raise ValueError('Undefined transform for dimension: %d' % (dim,))
+        img_dim = len(image.shape)
```

Comment:
Yes, actually I found this bug while writing the tests for the exceptions. Here I verify that the exception is raised when attempting to transform an image of dimension 2 and 4:
https://github.com/nipy/dipy/pull/700/files#diff-5a512c70f85666010a0c8c3d13db8f3aR508